File: 字串/165.比较版本号.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        i, j = 0, 0
        len1, len2 = len(version1), len(version2)
        while i < len1 or j < len2:
            num1 = 0
            while i < len1 and version1[i] != ".":
                num1 += num1 * 10 + int(version1[i])
                i += 1
            num2 = 0
            while j < len2 and version2[j] != ".":
                num2 += num2 * 10 + int(version2[j])
                j += 1
            i += 1
            j += 1
            if num1 != num2:
                return 1 if num1 > num2 else -1
        return 0

File: 字串/165.比较版本号.js
/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
//* 使用数字解决前导0的问题
var compareVersion = function(version1, version2) {
    let num1, num2;
    let p1 = 0, p2 = 0;
    let len1 = version1.length, len2 = version2.length;
    while (p1 < len1 || p2 < len2) {
        num1 = 0;
        while (p1 < len1 && version1[p1] !== '.') {
            num1 = num1 * 10 + +version1[p1];
            p1++;
        }
        num2 = 0;
        while (p2 < len2 && version2[p2] !== '.') {
            num2 = num2 * 10 + +version2[p2];
            p2++;
        }
        p1++;
        p2++;
        if (num1 !== num2) {
            return num1 > num2 ? 1 : -1;
        }
    }
    return 0
};